### PR TITLE
Shipping callback in NXO module

### PR DESCRIPTION
**Pull request by stephaniechiu**
### Summary of changes

This PR allows merchants to optionally create shipping callbacks by using the `onShippingChange` closure that's from the NXO SDK.

If a merchant wants to have shipping options, once they set the amount using `BTPayPalNativeCheckoutRequest(amount: "4.30")`, they will be able to call the `onShippingChange` closure and perform actions using the input parameters `change` and `action`.

The `onShippingChange` parameter was created in `BTPayPalNativeCheckoutRequest` like so `public var onShippingChange: ((PayPalCheckout.ShippingChange, PayPalCheckout.ShippingChangeAction) -> Void)?`

In `ShippingChangeAction`, a merchant can use `patch(request: PatchRequest)` `approve()`, `reject()`

Original PR - https://github.com/braintree/braintree_ios/pull/832

### Checklist

- [ ] Added a changelog entry

- stephaniechiu

### CODE REVIEW

File: Demo/Application/Features/PayPal - Native Checkout/BraintreeDemoPayPalNativeCheckoutViewController.swift [hunk 1]
```diff
@@ -1,10 +1,28 @@
 import Foundation
 import UIKit
 import BraintreePayPalNativeCheckout
+import PayPalCheckout
```
Comment (jaxdesmarais) [inline Demo/Application/Features/PayPal - Native Checkout/BraintreeDemoPayPalNativeCheckoutViewController.swift:4]: Agree with @jonathajones in previous PR that we do not want to require merchants to import `PayPalCheckout` to use specific features. Instead we should either create Braintree types that can be translated to the NXO types as or create a wrapper for the shipping callback types that can be passed through to the callback without the merchant having to pull in `PayPalCheckout` directly.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift [hunk 2]
```diff
@@ -2,12 +2,33 @@
 import BraintreePayPal
 #endif
 
-/// Options for the PayPal Checkout and PayPal Checkout with Vault flows.
+import PayPalCheckout
+
+/**
+ Options for the PayPal Checkout and PayPal Checkout with Vault flows.
+ */
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift:9]: Nit: update to /// doc strings.

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift [hunk 3]
```diff
@@ -2,12 +2,33 @@
...
-    let hermesPath: String = "v1/paypal_hermes/create_payment_resource"
+    /**
+     Initializes a PayPal Checkout request.
+
+     - Parameter amount: Used for a one-time payment. Amount must be greater than or equal to zero, may optionally contain exactly 2 decimal places separated by '.' and is limited to 7 digits before the decimal point.
+
+     - Parameter shippingCallback: Optional. Used for shipping changes and shipping change actions.
+
+     - Returns: A PayPal Checkout request.
+     */
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutRequest.swift:22]: Nit: update to /// doc strings.

**Comment by jaxdesmarais** [comment]
Can we add some details to the PR description around what this feature is? Not everyone on the Braintree team knows what shipping callbacks are and why/how a merchant would use shipping callbacks. This will allow for a better review process from Braintree team members that have not worked on/with native checkout previously.

File: Demo/Application/Features/PayPal - Native Checkout/BraintreeDemoPayPalNativeCheckoutViewController.swift (hunk 1 above)
Comment (stephaniechiu) [reply Demo/Application/Features/PayPal - Native Checkout/BraintreeDemoPayPalNativeCheckoutViewController.swift]: Thanks for the feedback! That makes sense that we don't want merchants to import `PayPalCheckout` directly. I'll make the changes accordingly

**Comment by stephaniechiu** [comment]
Closing until conflicts resolve

**Pull request closed without merging**